test/routes/editor_routes_test.rb

```diff
@@ -0,0 +1,34 @@
+class EditorRoutesTest < ActionDispatch::IntegrationTest
+  def setup
+    @site = create(:site)
+    @post = create(:post, :whatever, site: @site)
+    @category = create(:category, site: @site)
+    @fixed_page = create(:fixed_page, site: @site)
+    @link = create(:link, site: @site)
+    @credit_role = create(:credit_role, site: @site)
+    @participant = create(:participant, site: @site)
+    @serial = create(:serial, site: @site)
+    https!
+    host! @site.fqdn
+  end
+
+  data do
+    hash = {}
+    hash["/editor"] = [nil, "/editor", "/sites/%{site_fqdn}/editor"]
+    %w(post category fixed_page link credit_role participant serial).each do |target|
+      base_path = "/editor/#{target.pluralize}"
+      hash[base_path] = [target, base_path, "/sites/%{site_fqdn}#{base_path}"]
+      hash["#{base_path}/new"] = [target, "#{base_path}/new", "/sites/%{site_fqdn}#{base_path}/new"]
+      hash["#{base_path}/:id"] = [target, "#{base_path}/%{id}", "/sites/%{site_fqdn}#{base_path}/%{id}"]
+      hash["#{base_path}/:id/edit"] = [target, "#{base_path}/%{id}/edit", "/sites/%{site_fqdn}#{base_path}/%{id}/edit"]
+    end
+    hash
+  end
+  def test_redirect(data)
+    target, path, expected = data
+    target = instance_variable_get("@#{target}") if target
```

Comment:
No I can't.
These tests are exhaustive test.
It is too boring to extract code :boar: